File: Pascal-Delphi/Bank/AccountUnit.pas
unit AccountUnit;

{$MODE OBJFPC}
{$M+}
{$modeSwitch exceptions+}


interface

uses
  TransactionUnit;

type
  TAccount = class
  private
    FAccountHolder: string;
    FAccountNumber: string;
    FBalance: double;
  public
    constructor Create(AccountHolder: string; AccountNumber: string; Balance: double);
    function GetAccountHolder: string;
    function GetAccountNumber: string;
    function GetBalance: double;
    procedure Deposit(Amount: double);
    procedure Withdraw(Amount: double);
  end;

  TAccountList = class
  private
    FAccounts: array of TAccount;
    FCount: integer;
  public
    constructor Create;
    procedure AddAccount(Account: TAccount);
    function GetAccount(Index: integer): TAccount;
    function GetCount: integer;
    procedure ProcessTransaction(Transaction: TTransaction);
  end;

implementation

uses
  SysUtils;

{ TAccount }

constructor TAccount.Create(AccountHolder: string; AccountNumber: string;
  Balance: double);
begin
  FAccountHolder := AccountHolder;
  FAccountNumber := AccountNumber;
  FBalance := Balance;
end;

function TAccount.GetAccountHolder: string;
begin
  Result := FAccountHolder;
end;

function TAccount.GetAccountNumber: string;
begin
  Result := FAccountNumber;
end;

function TAccount.GetBalance: double;
begin
  Result := FBalance;
end;

procedure TAccount.Deposit(Amount: double);
begin
  FBalance := FBalance + Amount;
end;

procedure TAccount.Withdraw(Amount: double);
begin
  if Amount <= FBalance then
    FBalance := FBalance - Amount
  else
    raise Exception.Create('Insufficient funds.');
end;

{ TAccountList }

constructor TAccountList.Create;
begin
  FCount := 0;
end;

procedure TAccountList.AddAccount(Account: TAccount);
begin
  SetLength(FAccounts, FCount + 1);
  FAccounts[FCount] := Account;
  Inc(FCount);
end;

function TAccountList.GetAccount(Index: integer): TAccount;
begin
  Result := FAccounts[Index];
end;

function TAccountList.GetCount: integer;
begin
  Result := FCount;
end;

procedure TAccountList.ProcessTransaction(Transaction: TTransaction);
var
  I: integer;
  Account: TAccount;
begin
  for I := 0 to FCount - 1 do
  begin
    Account := FAccounts[I];
    if Account.GetAccountNumber = Transaction.GetAccountNumber then
    begin
      if Transaction.GetTransactionType = 'Deposit' then
        Account.Deposit(Transaction.GetAmount)
      else if Transaction.GetTransactionType = 'Withdrawal' then
        Account.Withdraw(Transaction.GetAmount);
      Break;
    end;
  end;
end;

end.

File: C#-Translated/Bank/Claude/Program3/AccountUnit.cs
using System;

namespace BankAccountSystem3
{
    public class TAccount
    {
        private string FAccountHolder;
        private string FAccountNumber;
        private double FBalance;

        public TAccount(string AccountHolder, string AccountNumber, double Balance)
        {
            FAccountHolder = AccountHolder;
            FAccountNumber = AccountNumber;
            FBalance = Balance;
        }

        public string GetAccountHolder()
        {
            return FAccountHolder;
        }

        public string GetAccountNumber()
        {
            return FAccountNumber;
        }

        public double GetBalance()
        {
            return FBalance;
        }

        public void Deposit(double Amount)
        {
            FBalance += Amount;
        }

        public void Withdraw(double Amount)
        {
            if (Amount <= FBalance)
                FBalance -= Amount;
            else
                throw new Exception("Insufficient funds.");
        }
    }

    public class TAccountList
    {
        private TAccount[] FAccounts;
        private int FCount;

        public TAccountList()
        {
            FCount = 0;
        }

        public void AddAccount(TAccount Account)
        {
            Array.Resize(ref FAccounts, FCount + 1);
            FAccounts[FCount] = Account;
            FCount++;
        }

        public TAccount GetAccount(int Index)
        {
            return FAccounts[Index];
        }

        public int GetCount()
        {
            return FCount;
        }

        public void ProcessTransaction(TTransaction Transaction)
        {
            for (int I = 0; I < FCount; I++)
            {
                TAccount Account = FAccounts[I];
                if (Account.GetAccountNumber() == Transaction.GetAccountNumber())
                {
                    if (Transaction.GetTransactionType() == "Deposit")
                        Account.Deposit(Transaction.GetAmount());
                    else if (Transaction.GetTransactionType() == "Withdrawal")
                        Account.Withdraw(Transaction.GetAmount());
                    break;
                }
            }
        }
    }
}